REL-592: release notes for 5.5.6 (#1697)
hazelcast/hz-docs: docs/modules/release-notes/pages/5-5-6.adoc

> = 5.5.5 Release Notes

Reviewer: ```suggestion
= 5.5.6 Release Notes
```

:description: These release notes list any new features, enhancements, fixes, and breaking changes implemented between version 5.5.5 and version 5.5.6 of Hazelcast Platform.

{description}

> **Release date**: 2025-05-16

Reviewer: ```suggestion
**Release date**: 2025-05-19
```

* **Fixed exception when using CREATE MAPPING with SQL statement-timeout-millis configured**: Resolved an issue where setting a non-zero value for `statement-timeout-millis` in the SQL configuration caused exceptions for statements that do not support timeouts, such as `CREATE MAPPING`. Previously, executing such statements would result in a descriptive but unnecessary exception (`CREATE MAPPING does not support timeout`), which could prevent client applications from starting if `CREATE MAPPING` was part of the initialization process. The fix ensures that non-timeout-supported statements like `CREATE MAPPING` succeed as expected, regardless of the `statement-timeout-millis` configuration.

> * **Fixed issue with hot backup failing after changing the backup folder during a rolling restart**: Resolved an issue where hot backups would fail if the underlying backup folder was changed during a rolling restart and a hot backup was subsequently initiated from the Management Center (MC). This fix ensures that hot backups function correctly even after the backup folder is updated during a rolling restart.

Reviewer: ```suggestion
* **Fixed issue with hot backup failing after changing the backup folder during a rolling restart**: Resolved an issue where hot backups would fail if the underlying backup folder was changed during a rolling restart and a hot backup was subsequently initiated from Management Center (MC). This fix ensures that hot backups function correctly even after the backup folder is updated during a rolling restart.
```

* **Fixed server exception caused by clients connecting to non-CP clusters with cp-direct-to-leader-routing enabled**: Resolved an issue where a client connecting to a non-CP cluster with `cp-direct-to-leader-routing: true` would trigger an exception in the server logs after 60 seconds. The exception (`HazelcastException: CP Subsystem is not enabled!`) caused unnecessary confusion for users. This fix ensures that clients with `cp-direct-to-leader-routing` enabled no longer trigger server exceptions when connected to non-CP clusters.

> * **Improved client error message for unsupported CP subsystem in Community Edition clients**: Resolved an issue where a Community Edition client attempting to interact with an Enterprise cluster using the CP subsystem would produce a misleading error message: "CP subsystem is a licensed feature. Please ensure you have an Enterprise license that enables CP." The error incorrectly suggested a licensing issue with the cluster, rather than indicating the actual problem with the client. The fix updates the error message to clearly indicate that the issue lies with the client, helping users identify and resolve the problem of using an incompatible client package more easily.

Reviewer 2: should we be using a variable for the community edition naming, same as we do for enterprise edition?

Reviewer: Yes, ideally

Reviewer: Updated

> * **Improved client error message for unsupported CP subsystem in Community Edition clients**: Resolved an issue where a Community Edition client attempting to interact with an Enterprise cluster using the CP subsystem would produce a misleading error message: "CP subsystem is a licensed feature. Please ensure you have an Enterprise license that enables CP." The error incorrectly suggested a licensing issue with the cluster, rather than indicating the actual problem with the client. The fix updates the error message to clearly indicate that the issue lies with the client, helping users identify and resolve the problem of using an incompatible client package more easily.

Reviewer: ```suggestion
* **Improved client error message for unsupported CP subsystem in {open-source-product-name} clients**: Resolved an issue where a {open-source-product-name} client attempting to interact with an {enterprise-product-name} cluster using the CP subsystem would produce a misleading error message: "CP subsystem is a licensed feature. Please ensure you have an Enterprise license that enables CP." The error incorrectly suggested a licensing issue with the cluster, rather than indicating the actual problem with the client. The fix updates the error message to clearly indicate that the issue lies with the client, helping users identify and resolve the problem of using an incompatible client package more easily.
```
